app/models/miq_event_definition.rb

```diff
@@ -1,10 +1,11 @@
 class MiqEventDefinition < ApplicationRecord
   include UuidMixin
 
-  validates_presence_of     :name
-  validates_uniqueness_of   :name
-  validates_format_of       :name, :with => /\A[a-z0-9_\-]+\z/i,
-    :allow_nil => true, :message => "must only contain alpha-numeric, underscore and hyphen characters without spaces"
+  validates :name, :uniqueness_when_changed => true,
+                   :presence                => true,
+                   :allow_nil               => true,
```

Comment:
to anyone looking at this line in the future, it was suggested that I leave it in in the spirit of changing as little as possible.

Reply:
and to anyone in the future
I think this line is unnecessary but remember there being a reason to do this in some cases and can't remember

So @d-m-u was more than patient as she submitted a pr, updated to remove it, and then updated when I asked her to add it back.